version: '3.8'

services:
  db:
    image: postgres:latest
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      DATABASE_NAME: pms
      DATABASE_USER: admin
      DATABASE_PASSWORD: adminpassword
      DATABASE_HOST: local
      DATABASE_PORT: 5432
    ports:
      - "54352:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $${DATABASE_USER} -d $${DATABASE_NAME}"]
      interval: 30s
      timeout: 10s
      retries: 5

  web:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    environment:
      DATABASE_NAME: pms
      DATABASE_USER: admin
      DATABASE_PASSWORD: adminpassword
      DATABASE_HOST: local
      DATABASE_PORT: 5432
    depends_on:
      db:
        condition: service_healthy
    
    healthcheck:
      test: ["CMD-SHELL", "curl -f http://localhost:8000/ || exit 1"]
      interval: 30s
      timeout: 10s
      retries: 5


volumes:
  postgres_data:
